### ISSUE TITLE
How can I play my Live Stream automatically using Clappr once the transmux is completed and the URL exists?

### ISSUE THREAD

**Issue by CodeBradley** [clappr/clappr#1950]
**What do you want to do with Clappr?**

**tl;dr** I want to figure out how to automatically play a live stream once it's started using React/Clappr.

I would like to stream from OBS to an RTMP Ingest server, transmux it to HLS (m3u8) using FFMPEG (Node Media Server) and then once the transmux is complete and the file/url exists automatically start playing the video in my browser using autoPlay/mute config options. What's the best way to go about this?

**What have you tried so far?**

I'm currently using a socket connection between the client/media server and sending the streamName (key) through the socket, creating the URL and attaching the player only once the source is set (which I have as a react state). Going this route the socket message starts the video player before the m3u8 file is served though so I have to add a 2.5s timeout after being notified the transmux is running which just seems like a sloppy way to go about it for me.

I'm also considering switching to just a server side event to notify the client and then once notified, the client makes an authenticated call to the server and then acquires the stream path (or generates the path without the server using the users localStorage data) and then attaching the player.

I'm also looking into how pull's work, but that's a new concept for me and haven't had time to look into it much at this point in time.

**Comment by joaopaulovieira** [comment]
Keep it open.

**Comment by CodeBradley** [comment]
I believe I solved this but don't have the code on this PC, if anyone's interested just lmk and I'll find my old PC and grab the solution.
